E2E fixes for Gutenberg 10.6.x / 10.7.x

fullofcaffeine opened the pull request:
Tracking issue: https://github.com/Automattic/wp-calypso/issues/52595, https://github.com/Automattic/wp-calypso/issues/53135.

Trigger the tests in TeamCity from this branch.

fullofcaffeine committed:
Remove expectation for __experimentalBlock as it has been deprecated

WunderBart commented:
Looks like there's only [1 test failing](https://teamcity.a8c.com/viewLog.html?buildId=6195314&buildTypeId=calypso_WPComTests_gutenberg_desktop) - the Premium Block is invalidating. This might be fixed via https://github.com/Automattic/jetpack/pull/19844 so let's wait until it's in. cc @fullofcaffeine @apeatling

fullofcaffeine committed:
Update E2E experimental guard to check for __unstableInserterMenuExtension

fullofcaffeine commented:
This can't be merged until >= 10.6 is on production, as it includes a test fix that's specific to that version (the `__experimental` to `__unstable` API change) and supporting both in the test wouldn't be worth the effort, I think.

fullofcaffeine committed:
Fix inserter CSS locator

fullofcaffeine committed:
Add deprecated fallback to make upgrade tests pass for now

fullofcaffeine committed:
Fix driver undefined driver instance issue

fullofcaffeine committed:
Fix inserter locator in `getShownBlockInserterItems()`

fullofcaffeine committed:
Fix inserter locators in `closeBlockInserter()`

fullofcaffeine committed:
Remove deprecated fallback as it is not strictly needed

This is not strictly needed for the suite for 10.7.x to pass, so let's avoid the overhead of having code that is not that useful laying around the tests.

fullofcaffeine commented on test/e2e/lib/gutenberg/gutenberg-editor-component.js:
```diff
@@ -387,8 +385,10 @@ export default class GutenbergEditorComponent extends AsyncBaseContainer {
 		);
 
+		// @todo Remove the deprecated query once GB v10.7+ is up on prod. This is needed for now to make sure all tests (more specifically upgrade tests that use non-edge sites) pass.
+		const insertedBlockItemDeprecatedQuery = `.edit-post-layout__inserter-panel .block-editor-block-types-list button.editor-block-list-item-${ prefix }${ blockClass }`;
```
Ops, this should have been named "inserter" not "inserted" 🤦🏻. It didn't affect the actual fix, but wasn't the correct name. Good thing is that we'll remove it now that v10.7.1 is on WPCOM.

fullofcaffeine replied:
Removed in https://github.com/Automattic/wp-calypso/pull/53395.
